drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/DRLVisitorImpl.java

```diff
@@ -193,7 +194,7 @@ public FunctionDescr visitFunctiondef(DRLParser.FunctiondefContext ctx) {
                 functionDescr.addParameter(typeTypeContext.getText(), variableDeclaratorIdContext.getText());
             });
         }
-        functionDescr.setBody(getTextPreservingWhitespace(ctx.block()));
+        functionDescr.setBody(stripBracesFromBlock(getTextPreservingWhitespace(ctx.drlBlock())));
```

Comment:
```suggestion
        functionDescr.setBody(getTextPreservingWhitespace(ctx.drlBlock().drlBlockStatement()));
```
@tkobayas by using block statements instead of the whole block, we could avoid "manually" trimming away the braces.

Of course, this requires a new whitespace-text-preserving function that works on a list of contexts, something like this:
```java
    public static String getTextPreservingWhitespace(List<? extends ParserRuleContext> ctx) {
        return ctx.stream().map(Antlr4ParserStringUtils::getTextPreservingWhitespace).collect(Collectors.joining());
    }
```
I find this simpler to understand and less error-prone (and maybe also reusable for other use cases) than the low-level string manipulation in `stripBracesFromBlock()`.

Wdyt?

Reply:
@yurloc Thank you! Applied the suggestion.